ui/src/SignUp.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import './SignUp.css';
+import axios from 'axios';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import InputGroup from 'react-bootstrap/InputGroup';
+import FormControl from 'react-bootstrap/FormControl';
+
+const validationSchema = Yup.object({
+    username: Yup.string().email().max(30).required('Required'),
+    password: Yup.string().min(8).max(50).matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,})/,
+    'Password must have 8 characters, 1 lowercase, 1 uppercase, 1 number, and 1 special character').required('Required'),
+    confirmPassword: Yup.string().required('Required').oneOf([Yup.ref('password'), null], 'Passwords must match')
+});
+
+function SignUp() {
+
+    const { handleSubmit, handleChange, values, errors } = useFormik({
+        initialValues: {
+            username: '',
+            password: '',
+            confirmPassword: ''
+        },
+        validationSchema,
+        onSubmit(values) {
+            axios.post('http://localhost:8080/users/sign-up', ({
+                username: values.username,
+                password: values.password
+            })).then(res => {
+                console.log(res);
```

Comment:
Should we send them to a different page after they sign up?

Reply:
I think we should redirect them to image upload